Data pool param not required anymore to create noobaa account(rpc)

response = mcg.send_rpc_query(
            api="account_api",
            method="create_account",
            params={
                "email": email,
                "name": name,
                "has_login": admin_access,
                "s3_access": s3_access,
                "default_pool": backingstore_name,
                "allowed_buckets": {
                    "full_permission": full_bucket_access,
                    "permission_list": buckets,
                },
            },
        ).json()
>       self.access_key_id = response["reply"]["access_keys"][0]["access_key"]
E       KeyError: 'reply'

--- a/ocs_ci/ocs/resources/bucket_policy.py
+++ b/ocs_ci/ocs/resources/bucket_policy.py
@@ -79,37 +79,23 @@ class NoobaaAccount(object):
         """
         self.account_name = name
         self.email_id = email
-        if float(config.ENV_DATA["ocs_version"]) < 4.9:
-            response = mcg.send_rpc_query(
-                api="account_api",
-                method="create_account",
-                params={
-                    "email": email,
-                    "name": name,
-                    "has_login": admin_access,
-                    "s3_access": s3_access,
-                    "default_pool": backingstore_name,
-                    "allowed_buckets": {
-                        "full_permission": full_bucket_access,
-                        "permission_list": buckets,
-                    },
-                },
-            ).json()
-        else:
-            response = mcg.send_rpc_query(
-                api="account_api",
-                method="create_account",
-                params={
-                    "email": email,
-                    "name": name,
-                    "has_login": admin_access,
-                    "s3_access": s3_access,
-                    "allowed_buckets": {
-                        "full_permission": full_bucket_access,
-                        "permission_list": buckets,
-                    },
-                },
-            ).json()
+        params_dict = {
+            "email": email,
+            "name": name,
+            "has_login": admin_access,
+            "s3_access": s3_access,
+            "default_pool": backingstore_name,
+            "allowed_buckets": {
+                "full_permission": full_bucket_access,
+                "permission_list": buckets,
+            },
+        }
+        params_dict if float(config.ENV_DATA["ocs_version"]) < 4.9 else params_dict.pop(
+            "default_pool"
+        )
+        response = mcg.send_rpc_query(
+            api="account_api", method="create_account", params=params_dict
+        ).json()
         self.access_key_id = response["reply"]["access_keys"][0]["access_key"]
         self.access_key = response["reply"]["access_keys"][0]["secret_key"]
         self.s3_endpoint = mcg.s3_endpoint

--- a/ocs_ci/ocs/resources/bucket_policy.py
+++ b/ocs_ci/ocs/resources/bucket_policy.py
@@ -3,6 +3,7 @@ import logging
 
 import boto3
 
+from ocs_ci.framework import config
 from ocs_ci.ocs import constants
 from ocs_ci.ocs.bucket_utils import retrieve_verification_mode
 
@@ -78,21 +79,37 @@ class NoobaaAccount(object):
         """
         self.account_name = name
         self.email_id = email
-        response = mcg.send_rpc_query(
-            api="account_api",
-            method="create_account",
-            params={
-                "email": email,
-                "name": name,
-                "has_login": admin_access,
-                "s3_access": s3_access,
-                "default_pool": backingstore_name,
-                "allowed_buckets": {
-                    "full_permission": full_bucket_access,
-                    "permission_list": buckets,
+        if float(config.ENV_DATA["ocs_version"]) < 4.9:
+            response = mcg.send_rpc_query(
+                api="account_api",
+                method="create_account",
+                params={
+                    "email": email,
+                    "name": name,
+                    "has_login": admin_access,
+                    "s3_access": s3_access,
+                    "default_pool": backingstore_name,
+                    "allowed_buckets": {
+                        "full_permission": full_bucket_access,
+                        "permission_list": buckets,
+                    },
                 },
-            },
-        ).json()
+            ).json()
+        else:
+            response = mcg.send_rpc_query(
+                api="account_api",
+                method="create_account",
+                params={
+                    "email": email,
+                    "name": name,
+                    "has_login": admin_access,
+                    "s3_access": s3_access,
+                    "allowed_buckets": {
+                        "full_permission": full_bucket_access,
+                        "permission_list": buckets,
+                    },
+                },
+            ).json()
         self.access_key_id = response["reply"]["access_keys"][0]["access_key"]
         self.access_key = response["reply"]["access_keys"][0]["secret_key"]
         self.s3_endpoint = mcg.s3_endpoint

